src/dbnode/storage/index.go

```diff
@@ -1436,3 +1440,230 @@ func (shards dbShards) IDs() []uint32 {
 	}
 	return ids
 }
+
+func (i *nsIndex) AggregateQuery(
```

Comment:
nit: refactor this so that `Query()` and `AggregateQuery()` share the structural components of the method.

Reply:
Was trying that but it got a little messy, will take another stab at it